src/providers/exploreState/utils.ts

```diff
@@ -46,7 +47,10 @@ export function initExploreState(
     ...INITIAL_STATE,
     catalogState: decodedCatalogParam,
     categoryGroupConfigs: entityPageState[entityListType].categoryGroupConfigs,
-    entityPageState: initEntityPageState(config),
+    entityPageState: updateEntityPageState(entityListType, entityPageState, {
```

Comment:
The `entityPageState` for the selected entity should be initialised with `filterCount` and `filterState` from the decoded filter param.